### ISSUE TITLE
state :valid breaks validations

### ISSUE THREAD

**Issue by reicheltd** [aasm/aasm#335]
aasm do
      state :valid
    end

gives me

`wrong number of arguments (1 for 0)`
`My_model#perform_validations`
`activerecord (4.2.5.2) lib/active_record/validations.rb, line 83`

    def perform_validations(options={}) # :nodoc:
      options[:validate] == false || valid?(options[:context])
    end

Using Rails 4.2.4.2 and Ruby 2.2.4

Maybe stupid of me to try to name a state `valid`, but I needed some time to figure it out :( Should be a warning somewhere. Thanks

**Comment by ervoiox** [comment]
It would be nice if you could specify a prefix to the public methods that AASM adds.

```ruby
class SomeModel
  aasm, state_prefix: 'aasm_state' do
    state :valid
  end
end

some_model.aasm_state_valid? # => true
```

**Comment by pirj** [comment]
@ervoiox It's possible to scope in a [slightly different way](https://github.com/aasm/aasm/blob/master/README.md#multiple-state-machines-per-class). 

Code that should show a warning message to protect original methods from being overridden doesn't work.

**Comment by alto** [comment]
I just released version `4.10.0` including the changes from #340.
